src/VariantGenerator/SetupResultInterface.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusImagePlugin\VariantGenerator;
+
+interface SetupResultInterface
+{
+    public function getGeneratorName(): string;
```

Comment:
Couldn't this return the actual generator instead of just the name?

Reply:
It could, but I feel that would be a weird "coupling" maybe? Like I don't need the concrete instance. If I want to act on the "results" I can resolve the generator some other way. But if you prefer it that way, I can change it.

Reply:
I believe it's already coupled when we have the generator name, but when we have the generator at the time of instantiation we could just as well pass the generator. It's better DX imo